src/testdir/test_functions.vim

```diff
@@ -940,6 +940,18 @@ func Test_mode()
   execute "normal! gR\<C-o>g@l\<Esc>"
   call assert_equal('n-niV', g:current_modes)
 
+  " Test statusline updates for overstike mode
+  CheckRunVimInTerminal
```

Comment:
It's better to either use an `if CanRunVimInTerminal()` block here or move this block to a separate function.